scripts/visualization/generate_map.py

```
@@ -70,25 +74,351 @@ def get_available_years(excel_path: Path) -> list[int]:
-                return max(full_years)
+            preferred_statuses = ["official_offer", "full"]
+            for status in preferred_statuses:
+                preferred_years = metadata[metadata["data_status"].eq(status)][
+                    "year"
+                ].dropna()
+                preferred_years = [int(year) for year in preferred_years.unique()]
+                preferred_years = [year for year in preferred_years if year in years]
+                if preferred_years:
+                    return max(preferred_years)
     except Exception:
         pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)